cpp/include/cudf/detail/copy_if_else.cuh

```diff
@@ -44,18 +44,19 @@ __launch_bounds__(block_size) __global__
                            mutable_column_device_view out,
                            size_type* __restrict__ const valid_count)
 {
-  size_type const tid            = threadIdx.x + blockIdx.x * block_size;
-  int const warp_id              = tid / warp_size;
+  auto tidx                      = cudf::detail::grid_1d::global_thread_id();
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**💡 suggestion:** ‏ This change could have an effect on performance. You should probably benchmark. Templating the block size rather than using blockDim.x is a classical kernel micro-optimization. It typically eliminates a register and enables some other compiler optimizations. Very valid when a kernel always uses the same block size.

You could add a template utility for this:

```
template <thread_index_type block_size>
grid_1d::global_thread_id() {
  return threadIdx.x + blockIdx.x * block_size;
}
```

Reply:
Done in 9e4154dd176c677ad7168c9f0770a7e2363c6970.